InnerEye/ML/configs/other/HelloContainer.py

```diff
@@ -47,16 +54,68 @@ def __len__(self) -> int:
+    is made on the whole dataset.
+    For cross validation (if required) we use k-fold cross-validation. The test set remains unchanged
+    while the training and validation data cycle through the k-folds of the remaining data.
     """
-    def __init__(self, root_folder: Path) -> None:
+    def __init__(
+            self,
+            root_folder: Path,
+            number_of_cross_validation_splits: int = 0,
+            cross_validation_split_index: int = 0) -> None:
         super().__init__()
-        self.train = HelloDataset(root_folder, start_index=0, end_index=50)
-        self.val = HelloDataset(root_folder, start_index=50, end_index=70)
-        self.test = HelloDataset(root_folder, start_index=70, end_index=100)
+        if number_of_cross_validation_splits <= 1:
+            # For 0 or 1 splits just use the default values on the whole data-set.
+            self.train = HelloDataset(root_folder, start_index=0, end_index=50)
+            self.val = HelloDataset(root_folder, start_index=50, end_index=70)
+            self.test = HelloDataset(root_folder, start_index=70, end_index=100)
+        else:
+            # Raise exceptions for unreasonable values
+            if cross_validation_split_index >= number_of_cross_validation_splits:
+                raise IndexError(f"The cross_validation_split_index ({cross_validation_split_index}) is too large "
+                f"given the number_of_cross_validation_splits ({number_of_cross_validation_splits}) requested")
+            raw_data = np.loadtxt(root_folder / "hellocontainer.csv", delimiter=",")
+            np.random.seed(42)
+            np.random.shuffle(raw_data)
+            if number_of_cross_validation_splits >= len(raw_data):
+                raise ValueError(f"Asked for {number_of_cross_validation_splits} cross validation splits from a "
+                f"dataset of length {len(raw_data)}")
+            # Hold out the last 30% as test data
+            self.test = HelloDataset.from_raw_data(raw_data[70:100])
+            # Create k-folds from the remaining 70% of the data-set. Use one for the validation
+            # data and the rest for the training data
+            raw_data_remaining = raw_data[0:70]
+            split_raw_data_remaining = np.array_split(raw_data_remaining, number_of_cross_validation_splits)
+            self.val = HelloDataset.from_raw_data(split_raw_data_remaining[cross_validation_split_index])
+            train: List[Any] = []
+            for train_index in range(number_of_cross_validation_splits):
+                if train_index != cross_validation_split_index:
+                    if len(train) > 0:
+                        train = np.append(train, split_raw_data_remaining[train_index], axis=0)
+                    else:
+                        train = split_raw_data_remaining[train_index]
+            self.train = HelloDataset.from_raw_data(train)
```

Comment:
Can we use a standard library function for that, like https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html?

Reply:
Done.